cardano-testnet/src/Testnet/Components/Query.hs

```diff
@@ -268,21 +268,21 @@ watchEpochStateUpdate
   :: forall m a. (HasCallStack, MonadIO m, MonadTest m, MonadAssertion m)
   => EpochStateView -- ^ The info to access the epoch state
   -> EpochInterval -- ^ The maximum number of epochs to wait
-  -> ((AnyNewEpochState, SlotNo, BlockNo) -> m (Maybe a)) -- ^ The guard function (@Just@ if the condition is met, @Nothing@ otherwise)
-  -> m (Maybe a)
+  -> ((AnyNewEpochState, SlotNo, BlockNo) -> m (LedgerStateCondition, a)) -- ^ The guard function (@Just@ if the condition is met, @Nothing@ otherwise)
```

Comment:
`LedgerStateCondition` needs to be renamed to a more general name. `ConditionCheckResult` perhaps? Will do in a follow-up PR to API.